newIDE/app/src/ResourcesList/ResourceUtils.js

```diff
@@ -80,7 +80,26 @@ export const copyAllToProjectFolder = (
         return resourcePath;
       }
 
-      const resourceBasename = path.basename(resourcePath);
+      let resourceBasename = path.basename(resourcePath),
+        splitAt = resourceBasename.lastIndexOf('.'),
+        fileName = resourceBasename.substring(0, splitAt),
+        fileExtension = resourceBasename.substring(splitAt),
+        fileNumber = 1;
+
+      const AllResourcesNames = project
+        .getResourcesManager()
+        .getAllResourceNames();
+
+      for (let itr = 0; itr < AllResourcesNames.size(); itr++) {
+        if (fileName + fileExtension === AllResourcesNames.at(itr)) {
+          fileName = resourceBasename.substring(0, splitAt) + fileNumber;
```

Comment:
calling `resourceBasename.substring(0, splitAt)` at every iteration seems unnecessary and to add complexity, making the whole thing run longer. You can just make a copy of fileName before the iteration to minimize the method calls:
```suggestion
      const originalFileName = fileName;

      for (let itr = 0; itr < AllResourcesNames.size(); itr++) {
        if (fileName + fileExtension === AllResourcesNames.at(itr)) {
          fileName = roriginalFileName + fileNumber;
```